src/containers/sound-editor.jsx

```diff
@@ -51,7 +59,11 @@ class SoundEditor extends React.Component {
             trimEnd: null
         });
     }
-    submitNewSamples (samples, sampleRate) {
+    submitNewSamples (samples, sampleRate, skipUndo) {
+        if (!skipUndo) {
```

Comment:
It's a good question, I think that because the editor may be extracted into a repo, the internal logic for undo/redo needs to be kept internal.